### ISSUE TITLE
Russian Localization

### ISSUE THREAD

**Issue by Koncord** [Interkarma/daggerfall-unity#2074]
Hello,
I'm working on Russian localization.

Do you plan to move all strings to the TextTables (months names for e.g.)?

I also have an issue with overriding fonts via 
```cs
fnt.LoadSDFFontAsset("Fonts/FONT0003-ru-RU-SDF");
var locId = new LocaleIdentifier(CultureInfo.GetCultureInfo("ru-RU"));
locale = LocalizationSettings.AvailableLocales.GetLocale(locId);
TextManager.Instance.RegisterLocalizedFont(locale, DaggerfallFont.FontName.FONT0003, fnt);
```

``LoadSDFFontAsset`` Throws an exception here: https://github.com/Interkarma/daggerfall-unity/blob/f481c12adf87a26820ad19136943c0dd3e41b8b7/Assets/Scripts/Game/UserInterface/DaggerfallFont.cs#L476-L478
due to ``atlas`` not being serialized. I have solved this issue by changing those lines to:
```cs
float atlasWidth = tmpFont.atlasTexture.width;
float atlasHeight = tmpFont.atlasTexture.height;
```
It also seems that I'll need to modify the code to support Russian better. I'll need to add support for different word endings, for example, strings like ``%s`` ("The %s's Odd" where %s is "Duke" for e.g)  cannot be translated without knowing the context usage.
I understand that localization support is still in pre-release.

**Comment by Interkarma** [comment]
Hi @Koncord! :) Yep, all text will be moved to string tables in stages. This stage is all about TEXT.RSC text and most of the "Internal" text (e.g. from FALL.EXE). I'm still writing first-pass tutorials for all of this and taking feedback from folks like yourself around any limitations found - which could be unique for each language.

The entire process will take a while. Daggerfall's text data is really messy and not planned out to start with, so a big part of the challenge is just trying to bring it all together in a fairly unified way. We will get there eventually so translation mods can cover all text displayed in-game.

Regarding fonts, Please update to 0.11.2 for some changes that help with this issue specifically. There's another workaround to set the older .atlas variable permanently via debugger view in Inspector, but that was too cumbersome and have already made the .atlasTexture change you suggest above in core. https://github.com/Interkarma/daggerfall-unity/commit/df5361e52b97b922b9e9b634eb056b7a82b21aae

Could you please help me understand the context issue with "The %s's Odd" a little better? I'm only an English speaker and don't easily understand the nuances of other languages. We can almost certainly work towards a solution for this once I understand better. I don't believe the text you're referring to has entered localization system yet and your feedback can help this improve later.

If you haven't already please check out the [Translation](https://forums.dfworkshop.net/viewforum.php?f=26) area on our forums. This is the de facto place where all the various DFU translation efforts can share experience and help each other.

**Comment by Koncord** [comment]
> Regarding fonts, Please update to 0.11.2

I'm on the 0.11.2. And crashing in Unity Player without editing the lines above.

> Could you please help me understand the context issue with "The %s's Odd" a little better?

In Russian, we have multiple word endings depends on grammatical gender/number/declension.

For example dates: 
"Январь" (January), 
can be shaped to "Января" when you mean date for eg "15 Января" (January 15th),
or, when you will answer on the question "when": "Когда? В Январе" (When? In January).
Another example is the item material: Leather, Steel, Daedric, etc. It must be translated differently for some parts of the armor.
Currently, I solved it via shortening words by shrinking ending at all.

I'm not familiar with Daggerfall internals yet. Are the title names used somewhere else? If not, I can translate them directly in the correct form.

As another solution, It can be done via using copies of all strings for example with "The %s's Odd" it can use one of the "Duke" in the table, specially generated for place names. Seems like it is a more unified solution for all languages.

**Comment by Interkarma** [comment]
I see what you mean - you are correct, I missed that. I'll roll that fix into 0.11.3.

Thank you for the extra details. I'm sure we can work out a solution for this when titles and some of these other elements enter localization.

**Comment by deepfighter** [comment]
@Koncord That's a typical problem, which exists since Classic. Daggerfall is not just translating text but also work with a hell lot of variables. To know where the game gets them is the key. Your "Duke" string is in the FALL.EXE, which Interkarma already moved to the text table. Some further names of places are in the MAPS.BSA, which quests or texts could refere as well, which can currently only be translated by a specific translation software for Classic. We had and still have to work with those problems for the German version. A lot of people will understand it anyway and fixing can be done later. If you are getting to quests, you will see, that it's merely impossible to make it 100% perfect, without changing the internal Daggerfall code. 

@Interkarma One solution, which could solve such problems - partially - would be, if we could add something behind the variables to work around it. Right now you need a special character like ' that the game recognizes that the variable (%s) should stay the same, regardless of any further addition. %s'en works, %sen (obviously) not. For example for quests it's difficult to know if the persons behind a variable are male or female. Depending on this, there will be some other grammatical usage of words, so usually you have to get with one way and hope that it will be correct.

**Comment by Interkarma** [comment]
Thanks for raising atlas issue. This has been fixed in 0.11.3. I can also see you've also moved over to discussion in forums, thank you. :)

I'll close this one now.
